Common/Model.cs

```diff
@@ -0,0 +1,143 @@
+using System;
+using System.Collections.Generic;
+using OpenTK.Graphics.OpenGL4;
+using OpenTK.Mathematics;
+using Assimp;
+using System.Drawing;
+using System.Drawing.Imaging;
+using PixelFormat = OpenTK.Graphics.OpenGL4.PixelFormat;
+using TextureWrapMode = OpenTK.Graphics.OpenGL4.TextureWrapMode;
+using AssimpMesh = Assimp.Mesh;
+
+namespace LearnOpenTK.Common
+{
+    public class Model
+    {
+        // model data
+        private List<Mesh> meshes;
+
+        // constructor, expects a filepath to a 3D model.
+        public Model(string path)
+        {
+            LoadModel(path);
+        }
+
+        // draws the model, and thus all its meshes
+        public void Draw()
+        {
+            for (int i = 0; i < meshes.Count; i++)
+                meshes[i].Draw();
+        }
+
+        // loads a model with supported ASSIMP extensions from file and stores the resulting meshes in the meshes list.
+
+        private void LoadModel(string path)
+        {
+            //Create a new importer
+            AssimpContext importer = new AssimpContext();
+
+            //This is how we add a logging callback 
+            LogStream logstream = new LogStream(delegate (String msg, String userData)
+            {
+                Console.WriteLine(msg);
+            });
+            logstream.Attach();
+
+            //Import the model. All configs are set. The model
+            //is imported, loaded into managed memory. Then the unmanaged memory is released, and everything is reset.
+            Scene scene = importer.ImportFile(path, PostProcessSteps.Triangulate);
+
+            // check for errors
+            if (scene == null || scene.SceneFlags.HasFlag(SceneFlags.Incomplete) || scene.RootNode == null)
+            {
+                Console.WriteLine("Unable to load model from: " + path);
+                return;
+            }
+            
+            //Reset the meshes and textures
+            meshes = new List<Mesh>();
```

Comment:
Looking at this a little closer, it looks like the comments were a bit misleading. Since this is part of the LoadModel method within the constructor, we need to define the empty list here so that we can add Meshes to is within the ProcessNode method. Therefore we would never need to Clear it since it is only created and added to during the constructor.

Reply:
I guess we can leave it as it is.